projects/dot-custom-elements/src/components/dot-checkbox/dot-checkbox.tsx

```diff
@@ -0,0 +1,58 @@
+import { Component, Prop, State, Event, EventEmitter } from '@stencil/core';
+import { generateId } from '../../utils/utils';
+import Fragment from 'stencil-fragment';
+
+@Component({
+    tag: 'dot-checkbox',
+    styleUrl: 'dot-checkbox.scss'
+})
+export class DotCheckboxComponent {
+    @Prop() label: string;
+    @Prop() hint: string;
+    @Prop() options: string;
+    @Prop() value: string;
+    @Event() onCallback: EventEmitter;
+
+    @State() _options: any;
+    @State() _value: string;
+    _label: string;
+    _values = {};
+
+    componentWillLoad() {
+        this._options = this.options
+            .replace(/(\\r\\n|\\n|\\r)/gi, '|')
+            .split('|')
+            .filter((item) => item.length > 0);
+        this._label = `dotCheckbox_${generateId()}`;
+    }
+
+    setValue(event): void {
+        const checkBoxVal = event.target.value.toString();
+        this._values = { ...this._values, [checkBoxVal]: !this._values[checkBoxVal] };
+        this.onCallback.emit({ value: this._values });
```

Comment:
I don't understand what is this method doing, doc?

Reply:
this is going to change, so no need to change now